datafusion/sql/src/expr/identifier.rs

```diff
@@ -142,9 +135,7 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                             }
                         }
                     }
-                    not_impl_err!(
-                        "Compound identifiers not supported by ExprPlanner: {ids:?}"
-                    )
+                    exec_err!("could not parse compound identifier from {ids:?}")
```

Comment:
I think perhaps this should this `plan_err!` rather than `exec_err!` as it happens during planning?

Reply:
got it